cohttp-lwt/src/s.ml

```diff
@@ -21,9 +21,11 @@ end
     and close the resulting channels to clean up. *)
 module type Net = sig
   module IO : IO
-  type ctx [@@deriving sexp_of]
-  val default_ctx: ctx
-  val connect_uri : ctx:ctx -> Uri.t -> (IO.conn * IO.ic * IO.oc) Lwt.t
+
+  type resolvers
```

Comment:
This does need to be sexp-serialisable as well.

Reply:
It will be an opaque sexp-serialisable value where `resolvers` is a map of functions. It's fine for you?